R/Rules-methods.R

```diff
@@ -1001,6 +1001,295 @@ setMethod(
+    # equal to the target and calculate how often a dose is selected as MTD
+    # across iterations.
+    # The first element of the vector is the relative frequency that no
+    # dose in the grid is below or equal to the target, the
+    # second element that the 1st dose is the MTD, etc.
+    prob_mtd_lte <- table(factor(apply(prob_samples <= nextBest@target, 1, sum),
+                                  levels = 0:data@nGrid)) / size(samples)
+
+    # Handling of first element of the vector according to the selected method.
+    # If method = min: first dose = sum of no dose is below target and minimum
+    #   dose is below target
+    # If method = max: last dose = sum of no dose is below target and maximum
+    #   dose is below target
+    allocation_crit <- as.vector(prob_mtd_lte)
+    names(allocation_crit) <- as.character(c(0, data@doseGrid))
+
+    # In case that placebo is used, placebo and no dose are merged
+    if (data@placebo) {
+      allocation_crit[1] <- sum(allocation_crit[1:2])
+      allocation_crit <- allocation_crit[-2]
+    }
+
+    if (nextBest@method == "min") {
+      allocation_crit[2] <- sum(allocation_crit[1:2])
+    } else if (nextBest@method == "max") {
+      allocation_crit[length(allocation_crit)] <-
+        sum(allocation_crit[1], tail(allocation_crit, 1))
+    }
+    allocation_crit <- allocation_crit[-1]
+
+    # Determine the dose with the highest relative frequency
+    allocation_crit_dose <- as.numeric(names(allocation_crit))
+    dose_target <- allocation_crit_dose[which.max(allocation_crit)]
+
+    # Determine next dose
+    doses_eligible <- h_next_best_eligible_doses(
+      data@doseGrid,
+      doselimit,
+      data@placebo
+    )
+    next_dose_level_eligible <- which.min(abs(doses_eligible - dose_target))
+    next_dose <- doses_eligible[next_dose_level_eligible]
+
+    # Create a plot.
+    plt_data <- if (data@placebo && data@doseGrid[1] == next_dose) {
+      data.frame(x = as.factor(data@doseGrid),
+                 y = c(0, as.numeric(allocation_crit)) * 100)
+    } else {
+      data.frame(x = as.factor(allocation_crit_dose),
+                 y = as.numeric(allocation_crit) * 100)
+    }
+
+    doselimit_factor <- if (sum(allocation_crit_dose == doselimit) > 0) {
+      which(allocation_crit_dose == doselimit)
+    } else {
+      ifelse(test = data@placebo && data@doseGrid[1] == next_dose,
+        yes = 1.5,
+        no = sum(allocation_crit_dose < doselimit) + 0.5
+      )
+    }
+
+    p <- ggplot(
+      data = plt_data,
+      fill = "grey50",
+      colour = "grey50"
+    ) +
+      geom_col(aes(x, y), fill = "grey75") +
+      scale_x_discrete(drop = FALSE, guide = guide_axis(check.overlap = TRUE)) +
+       geom_vline(
+        xintercept = as.factor(dose_target),
+        lwd = 1.1,
+        colour = "black"
+      ) +
+      geom_text(
+        data = data.frame(x = as.factor(dose_target)),
+        aes(.data$x, 0),
+        label = "Est",
```

Comment:
what dose 'Est' stands for ?

Reply:
Estimated. I adpted this language from the other methods, e.g. NextBestMTD

Reply:
thanks for clarifying